pkg/cmd/dev/doctor.go

```diff
@@ -11,22 +11,64 @@
 package main
 
 import (
-	"errors"
+	"context"
+	"io/ioutil"
 	"log"
-	"os/exec"
+	"os"
+	osexec "os/exec"
 	"path/filepath"
 	"runtime"
 	"strconv"
 	"strings"
 
+	"github.com/cockroachdb/cockroach/pkg/cmd/dev/io/exec"
+	"github.com/cockroachdb/errors"
 	"github.com/spf13/cobra"
 )
 
+const devStatusFile = "bin/.dev-status"
+
+// devStatusVersion is the current "version" of the status checks performed by
+// `dev doctor``. Increasing it will force doctor to be re-run before other dev
+// commands can be run.
+const devStatusVersion = 1
+
+func checkDoctorStatus(ctx context.Context, ex *exec.Exec) error {
+	statusFile, err := pathInWorkdir(ctx, ex, devStatusFile)
+	if err != nil {
+		return err
+	}
+	content, err := ioutil.ReadFile(statusFile)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			content = []byte("0")
+		} else {
+			return err
+		}
+	}
+	status, err := strconv.Atoi(strings.TrimSpace(string(content)))
+	if err != nil {
+		return err
+	}
+	if status < devStatusVersion {
+		return errors.Errorf("Please run `dev setup` to refresh dev status, then try again.")
```

Comment:
I have no problem with the alias, but can we please make the error message refer to `dev doctor` (which is currently the canonical name of the command)? If we want to rename it I would prefer to keep it consistent and do a pass over the entire repo and docs to make sure that we use `status` instead of `doctor`.

Reply:
So I think it makes sense, if we're _demanding_ you run something to set up state to build, to refer to it as "setup" -- we want you to run setup to set things up to proceed.

If you're trying to inspect/resolve an issue, I have no problem with referring to it as "doctor" so I didn't feel the need to rename it everywhere? mostly just in this context that it felt like it should be callable as "setup"

Reply:
I'm not terribly passionate about this, but I think there is something to be said for not confusing folks about what the intended workflow is. We've been saying `dev doctor` forever -- I think folks are going to be confused/concerned when they hear about `dev setup` as a separate command, wonder if they've been missing a step all this time, etc. since there's no way to know that they're aliases without going out of your way to check. Ideally there shouldn't be any confusion.

Reply:
I just changed it to doctor for now.